aspnetcore/host-and-deploy/proxy-load-balancer/6.1samples/WebPS/Program.cs

```diff
@@ -0,0 +1,468 @@
+    });
+}
+
+var app = builder.Build();
+
+app.UseForwardedHeaders();
+
+if (!app.Environment.IsDevelopment())
+{
+    app.UseExceptionHandler("/Error");
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+app.UseStaticFiles();
+
+app.UseAuthorization();
+
+app.MapRazorPages();
+
+app.Run();
+#endregion
+#elif AZ
+#region snippet_az
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddRazorPages();
+builder.Services.AddCertificateForwarding(options =>
+    options.CertificateHeader = "X-ARR-ClientCert");
+
+var app = builder.Build();
+
+app.UseCertificateForwarding();
+
+if (!app.Environment.IsDevelopment())
+{
+    app.UseExceptionHandler("/Error");
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+app.UseStaticFiles();
+
+app.UseAuthorization();
+app.UseAuthentication();
+
+app.MapRazorPages();
+
+app.Run();
+#endregion
+#elif OWP
+#region snippet_owp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddRazorPages();
+builder.Services.AddCertificateForwarding(options =>
+    options.CertificateHeader = "YOUR_CERTIFICATE_HEADER_NAME");
+
+var app = builder.Build();
+
+app.UseCertificateForwarding();
+
+if (!app.Environment.IsDevelopment())
+{
+    app.UseExceptionHandler("/Error");
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+app.UseStaticFiles();
+
+app.UseAuthorization();
+app.UseAuthentication();
+
+app.MapRazorPages();
+
+app.Run();
+#endregion
+#elif OWP2
+#region snippet_owp2
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddRazorPages();
+builder.Services.AddCertificateForwarding(options =>
+{
+    options.CertificateHeader = "YOUR_CUSTOM_HEADER_NAME";
+    options.HeaderConverter = (headerValue) =>
+    {
+        // Conversion logic to create an X509Certificate2.
+        var clientCertificate = ConversionLogic.CreateAnX509Certificate2();
+        return clientCertificate;
+    };
+});
+
+var app = builder.Build();
+
+app.UseCertificateForwarding();
+
+if (!app.Environment.IsDevelopment())
+{
+    app.UseExceptionHandler("/Error");
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+app.UseStaticFiles();
+
+app.UseAuthorization();
+app.UseAuthentication();
+
+app.MapRazorPages();
+
+app.Run();
+#endregion
+
+public static class ConversionLogic
+{
+    public static System.Security.Cryptography.X509Certificates.X509Certificate2
+        CreateAnX509Certificate2() => new System.Security.Cryptography
+        .X509Certificates.X509Certificate2("abc");
+}
+#elif TRB
+#region snippet_trb
+using Microsoft.AspNetCore.HttpOverrides;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddRazorPages();
+builder.Services.Configure<ForwardedHeadersOptions>(options =>
+{
+    options.ForwardedHeaders =
+        ForwardedHeaders.XForwardedFor | ForwardedHeaders.XForwardedProto;
+});
+
+var app = builder.Build();
+
+app.UseForwardedHeaders();
+
+app.Run(async (context) =>
+{
+    context.Response.ContentType = "text/plain";
```

Comment:
We should look at replacing this sample with the new Request Logging Middleware (in a separate issue/PR).

Reply:
> We should look at replacing this sample with the new Request Logging Middleware (in a separate issue/PR).

Created #24530 to track this. Can you point me to a sample I can use? I assume that's for the next sample too.